cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesService.java

```diff
@@ -155,7 +168,8 @@ private void setConfig(ProfileStore profileStore, PreferencesTable preferencesTa
     }
   }
 
-  private void deleteConfig(EntityId entityId) {
+  private void deleteConfig(EntityId entityId) throws Exception {
+    authorizationEnforcer.enforce(entityId, authenticationContext.getPrincipal(), Action.EXECUTE);
     TransactionRunners.run(transactionRunner, context -> {
       PreferencesTable dataset = new PreferencesTable(context);
       Map<String, String> oldProp = dataset.getPreferences(entityId).getProperties();
```

Comment:
Same comment as above regarding moving the enforcement check downwards.

Reply:
same here